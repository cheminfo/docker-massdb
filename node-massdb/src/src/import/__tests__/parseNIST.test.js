'use strict';

let join = require('path').join;

let parseNIST = require('../parseNIST');

test('test NIST parsing', async () => {
  let file = join(__dirname, 'data/nist');
  let result = await parseNIST(file);
  expect(Object.keys(result)).toHaveLength(2);
  expect(result).toMatchSnapshot();
});
