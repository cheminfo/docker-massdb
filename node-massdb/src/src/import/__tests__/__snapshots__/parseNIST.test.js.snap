// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test NIST parsing 1`] = `
Array [
  Object {
    "data": Object {
      "x": Array [
        1,
        2,
      ],
      "y": Array [
        21,
        999,
      ],
    },
    "mf": "H2",
    "molfile": "Hydrogen
  -NISTMS-12099911132D 1   1.0         0.0          

  2  1  0  0  0  0  0  0  0  0  0
   -0.4060    0.6090    0.0000 H   0  0  0     0  0  0  0  0  0
    0.4060    0.6090    0.0000 H   0  0  0     0  0  0  0  0  0
  1  2  1  0  0  0  0
$$$$
",
    "name": "Hydrogen",
    "nist": "245692",
    "rn": "1333-74-0",
    "source": "American Petroleum Institute Research Project 44",
    "synonyms": Array [
      "Dihydrogen",
      "o-Hydrogen",
      "p-Hydrogen",
      "Molecular hydrogen",
      "H2",
      "UN 1049",
      "UN 1966",
    ],
  },
  Object {
    "data": Object {
      "x": Array [
        2,
        4,
      ],
      "y": Array [
        15,
        999,
      ],
    },
    "mf": "D2",
    "molfile": "Deuterium
  -NISTMS-12099911132D 1   1.0         0.0          

  2  1  0  0  0  0  0  0  0  0  0
   -0.4060    0.6090    0.0000 H   1  0  0     0  0  0  0  0  0
    0.4060    0.6090    0.0000 H   1  0  0     0  0  0  0  0  0
  1  2  1  0  0  0  0
$$$$
",
    "name": "Deuterium",
    "nist": "61316",
    "rn": "7782-39-0",
    "source": "D.HENNEBERG, MAX-PLANCK INSTITUTE, MULHEIM, WEST GERMANY",
    "synonyms": Array [
      "D2",
      "UN 1957",
    ],
  },
]
`;
